edx_proctoring/api.py

```diff
@@ -1424,7 +1424,7 @@ def _get_timed_exam_view(exam, context, exam_id, user_id, course_id):
     attempt_status = attempt['status'] if attempt else None
     has_due_date = True if exam['due_date'] is not None else False
     if not attempt_status:
-        if has_due_date_passed(exam['due_date']):
+        if has_due_date_passed(exam['due_date']) and not _is_self_paced_course(user_id, course_id):
```

Comment:
@afzaledx I think the approach we should take here is to use the due_date from the context object instead.
```
if has_due_date_passed(context['due_date']):
```
We will also want to replace all references to `exam['due_date']` with `context['due_date']` in this function.